src/ig/IGExporter.ts

```diff
@@ -836,10 +836,16 @@ export class IGExporter {
       reference: { reference: referenceKey }
     };
     if (pkgResource instanceof InstanceDefinition) {
+      const title = typeof pkgResource.title === 'string' ? pkgResource.title : null;
```

Comment:
I am seeing this pattern in enough places that I'd suggest creating a simple function like:
```ts
function stringOrNull(value: string): string | null {
  return typeof value === 'string' ? value : null;
}
```
Then you don't have to define intermediate variables like `title`, `name`, or `description` and instead just use that function where you need to reference them. E.g., instead of `title ??` on line 843 below, you'd have `stringOrNull(pkgResource.title) ??`.